bin/millau/runtime/src/rialto_messages.rs

```diff
@@ -16,21 +16,41 @@
 
 //! Everything required to serve Millau <-> Rialto message lanes.
 
-use bridge_runtime_common::messages;
+use crate::Runtime;
 
 use bp_message_lane::{
 	source_chain::TargetHeaderChain,
 	target_chain::{ProvedMessages, SourceHeaderChain},
-	InboundLaneData, LaneId, Message, MessageNonce,
+	InboundLaneData, LaneId, Message, MessageKey, MessageNonce,
 };
 use bp_runtime::InstanceId;
-use bridge_runtime_common::messages::MessageBridge;
+use bridge_runtime_common::messages::{self, MessageBridge};
 use frame_support::{
+	storage::generator::StorageMap,
 	weights::{Weight, WeightToFeePolynomial},
 	RuntimeDebug,
 };
+use pallet_message_lane::{DefaultInstance, InboundLanes, OutboundLanes, OutboundMessages};
+use sp_core::storage::StorageKey;
 use sp_trie::StorageProof;
 
+/// Storage key of the Millau -> Rialto message in the runtime storage.
+pub fn message_key(lane: &LaneId, nonce: MessageNonce) -> StorageKey {
+	let message_key = MessageKey { lane_id: *lane, nonce };
+	let raw_storage_key = OutboundMessages::<Runtime, DefaultInstance>::storage_map_final_key(message_key);
```

Comment:
Why isn't there a `RIALTO_BRIDGE_INSTANCE -> Instance` mapping? All 3 functions added here could stay completely generic over the instance and there would be just single match instead of 3.

Reply:
basically we would need:
```
fn dispatch_instance<F, I, R>(id: InstanceId, f: F, params: P) -> R where F: FnOnce<I>(P) -> R, I: Instance {...}
```